### ISSUE TITLE
infinite loop in TokenReplacingReader

### ISSUE THREAD

**Issue by jcsahnwaldt** [jjenkov/TokenReplacingReader#1]
[while(data != '}')](https://github.com/jjenkov/TokenReplacingReader/blob/master/src/main/java/com/jenkov/io/TokenReplacingReader.java#L49) will never stop if the source contains an unclosed token.

**Comment by jjenkov** [comment]
You are right! There there should be added a && data != -1 to the while clause, like this:

while(data != '}' && data != -1)
